Applications/Spire/Source/Ui/IconButton.cpp

```diff
@@ -46,25 +46,20 @@ IconButton::IconButton(QImage icon, Style style, QWidget* parent)
 }
 
 void IconButton::keyPressEvent(QKeyEvent* event) {
-  if(!event->isAutoRepeat() && (event->key() == Qt::Key_Enter ||
-      event->key() == Qt::Key_Return)) {
-    event->accept();
-    setDown(true);
-    Q_EMIT pressed();
-    return;
+  event->accept();
+  if(!event->isAutoRepeat()) {
+    switch(event->key()) {
+      case Qt::Key_Enter:
+      case Qt::Key_Return:
+      case Qt::Key_Space:
+        Q_EMIT clicked(isChecked());
+        return;
+    }
   }
-  QAbstractButton::keyPressEvent(event);
 }
 
 void IconButton::keyReleaseEvent(QKeyEvent* event) {
-  if(!event->isAutoRepeat() && isDown()) {
-    event->accept();
-    setDown(false);
-    Q_EMIT released();
-    Q_EMIT clicked(isChecked());
-    return;
-  }
-  QAbstractButton::keyReleaseEvent(event);
```

Comment:
You had better keep QAbstractButton::keyReleaseEvent(event), because a repeat timer is stopped inside the function.
Below is the code snippet of the function QAbstractButton::keyReleaseEvent:
void QAbstractButton::keyReleaseEvent(QKeyEvent *e)
{
    Q_D(QAbstractButton);

    if (!e->isAutoRepeat())
        d->repeatTimer.stop();

    switch (e->key()) {
    case Qt::Key_Select:
    case Qt::Key_Space:
        if (!e->isAutoRepeat() && d->down)
            d->click();
        break;
    default:
        e->ignore();
    }
}

Reply:
The repeatTimer is started when the setDown(bool) function is called, so in the case of the overridden key events I made a decision to not call setDown(...) to simplify the implementation. Because the timer isn't starting now on key presses, there's no need to have a corresponding call to setDown(false) to stop it.

The downside to this is that the IconButton will no longer emit pressed and released signals for key presses, but since we don't currently use them I decided to leave the call to setDown(...) out.

My goal was to make the intent clear, so that someone reading this knows that the key events are wholly the responsibility of the IconButton class, but I suppose it could be debated if that's desirable or note.

Reply:
Good, good, keep it.